### PR TITLE
Updated XDR filtered by status

### PR DESCRIPTION
**Pull request by yaakovpraisler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/33397

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CortexXDR/ReleaseNotes/2_9_1.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Palo Alto Networks Cortex XDR - Investigation and Response
+- Updated the option to filtered by the incident status
```
Comment (roysagi) [inline Packs/CortexXDR/ReleaseNotes/2_9_1.md:4]: Try looking at the RN templates and use them please: https://xsoar.pan.dev/docs/documentation/release-notes

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 2]
```diff
@@ -2935,6 +2980,8 @@ def main():
     base_url = urljoin(demisto.params().get('url'), '/public_api/v1')
     proxy = demisto.params().get('proxy')
     verify_cert = not demisto.params().get('insecure', False)
+    statuses = demisto.params().get('status')
```
Comment (roysagi) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:2983]: Does this automatically arrive as an array?

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 3]
```diff
@@ -2486,8 +2519,14 @@ def fetch_incidents(client, first_fetch_time, integration_instance, last_run: di
     if incidents_from_previous_run:
         raw_incidents = incidents_from_previous_run
     else:
-        raw_incidents = client.get_incidents(gte_creation_time_milliseconds=last_fetch,
-                                             limit=max_fetch, sort_by_creation_time='asc')
+        if len(statuses) > 0:
+            raw_incidents = []
+            for status in statuses:
+                raw_incidents += client.get_incidents(gte_creation_time_milliseconds=last_fetch, status=status,
```
Comment (roysagi) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:2525]: seems you support multi statuses in the command, why do we need the for loop then?

**Review by roysagi — CHANGES_REQUESTED** [review]
Seems good, see my notes.
Also, update the readme accordingly.
@reutshal - I'd like your opinion here as well.

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 4]
```diff
@@ -72,6 +72,20 @@ configuration:
   name: proxy
   required: false
   type: 8
+- additionalinfo: The status according to which the incidents will be filtered
+  display: Incident status
```
[the remark this answers is not in the document]
Comment (yaakovpraisler) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:76]: Done.

File: Packs/CortexXDR/ReleaseNotes/2_9_1.md (hunk 1 above)
Comment (yaakovpraisler) [reply Packs/CortexXDR/ReleaseNotes/2_9_1.md]: Done.

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 5]
```diff
@@ -72,6 +72,20 @@ configuration:
   name: proxy
   required: false
   type: 8
+- additionalinfo: 'The statuses of the incidents that will be fetched. Note: an incident whose status was changed to a filtered status after its creation time will not be fetched.'
```
Comment (anara123) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:75]: As part of the comment also explain that if no status is provided then incidents will all the statuses will be fetched

File: Packs/CortexXDR/ReleaseNotes/2_9_2.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+#### Integrations
+##### Palo Alto Networks Cortex XDR - Investigation and Response
+- Added the *status* integration parameter which enables you to filter fetched incidents by their status.
```
Comment (anara123) [inline Packs/CortexXDR/ReleaseNotes/2_9_2.md:3]: also added status argument to the `xdr-get-incidents` command

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 7]
```diff
@@ -2486,8 +2519,14 @@ def fetch_incidents(client, first_fetch_time, integration_instance, last_run: di
     if incidents_from_previous_run:
         raw_incidents = incidents_from_previous_run
     else:
-        raw_incidents = client.get_incidents(gte_creation_time_milliseconds=last_fetch,
-                                             limit=max_fetch, sort_by_creation_time='asc')
+        if len(statuses) > 0:
```
Comment (anara123) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:2522]: you should check for `statuses` for not being None. Otherwise this line will fail

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 8]
```diff
@@ -207,6 +208,13 @@ def get_incidents(self, incident_id_list=None, lte_modification_time=None, gte_m
                 'value': gte_creation_time_milliseconds
             })
 
+        if status:
+            filters.append({
+                'field': 'status',
+                'operator': 'eq',
+                'value': status
+            })
```
Comment (anara123) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:216]: can we do single request to fetch incidents with multiple statuses ?

**Review by roysagi — COMMENTED** [review]
@yaakovpraisler - Please add a unit test for the change in fetch incidents.

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py (hunk 8 above)
Comment (yaakovpraisler) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:216]: No, the only operators we have are `eq` and `neq`, we do not have an `in` operator.

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py (hunk 7 above)
Comment (yaakovpraisler) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:2522]: By default, `statuses` is an empty list.

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml (hunk 5 above)
Comment (yaakovpraisler) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:75]: Done.

File: Packs/CortexXDR/ReleaseNotes/2_9_2.md (hunk 6 above)
Comment (yaakovpraisler) [reply Packs/CortexXDR/ReleaseNotes/2_9_2.md:3]: Done.

File: Packs/CortexXDR/ReleaseNotes/2_9_2.md [hunk 9]
```diff
@@ -1,5 +1,6 @@
-
 #### Integrations
```
Comment (roysagi) [inline Packs/CortexXDR/ReleaseNotes/2_9_2.md:1]: @yaakovpraisler you are editing an existing RN! Please revert this change and add a new file and update the pack version

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 10]
```diff
@@ -1234,21 +1244,42 @@ def get_incidents_command(client, args):
...
+                gte_creation_time=gte_creation_time,
+                sort_by_creation_time=sort_by_creation_time,
+                sort_by_modification_time=sort_by_modification_time,
+                page_number=page,
+                limit=limit,
+                status=status
+            )
+
+        while len(raw_incidents) > limit:
+            raw_incidents.pop()
```
Comment (roysagi) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:1270]: why not simply:
```
if len(raw_incidents) > limit:
   raw_incidents[:limit]
```

while seems redundant.

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py (hunk 7 above)
Comment (roysagi) [reply Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:2522]: still, good practice to change this to `if statuses:` to cover all bases.

**Review by roysagi — CHANGES_REQUESTED** [review]
@yaakovpraisler - please fix the RN file here.

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 11]
```diff
@@ -1226,6 +1234,8 @@ def get_incidents_command(client, args):
     if since_creation_time:
         gte_creation_time, _ = parse_date_range(since_creation_time, TIME_FORMAT)
 
+    statuses = args.get('status', '').split(',')
```
Comment (reutshal) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:1237]:
```suggestion
    statuses = argToList(args.get('status', ''))
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 12]
```diff
@@ -164,6 +179,21 @@ script:
       name: limit
       required: false
       secret: false
+    - auto: PREDEFINED
+      default: false
+      description: Filters only incidents in the specified status.
+      isArray: true
+      name: status
+      predefined:
```
Comment (reutshal) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:187]: Since this is a list argument, pre-defined values will give the impression of a single select argument (you can try it in the cli, you will be able to choose only one option from the list). I would make it a simple text argument and supply all the options in the argument's description. @roysagi It's your call.

**Review by roysagi — CHANGES_REQUESTED** [review]

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 13]
```diff
@@ -2486,8 +2520,14 @@ def fetch_incidents(client, first_fetch_time, integration_instance, last_run: di
     if incidents_from_previous_run:
         raw_incidents = incidents_from_previous_run
     else:
-        raw_incidents = client.get_incidents(gte_creation_time_milliseconds=last_fetch,
-                                             limit=max_fetch, sort_by_creation_time='asc')
+        if statuses:
+            raw_incidents = []
+            for status in statuses:
+                raw_incidents += client.get_incidents(gte_creation_time_milliseconds=last_fetch, status=status,
+                                                      limit=max_fetch, sort_by_creation_time='asc')
```
Comment (roysagi) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:2527]: one last thing that just came to my mind:

the fetch_incidents counts on the `raw_incidents` variable to be sorted by the creation time of the incidents, but because you now fetch in batches by status, this could take the variable out of order. We need to add a resorting after all the incidents are gathered.

```raw_incidents = sorted(raw_incidents, key = lambda inc: inc['creation_time'])```

**Review by roysagi — CHANGES_REQUESTED** [review]

**Review by roysagi — APPROVED** [review]
